stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -151,6 +150,7 @@ public protocol ActorIdentity: Sendable, Hashable, Codable {}
 
 @available(SwiftStdlib 5.6, *)
 public struct AnyActorIdentity: ActorIdentity, @unchecked Sendable, CustomStringConvertible {
+  // FIXME: This probably shouldn't be public
   public let underlying: Any
```

Comment:
weirdly enough, it has to be public or we have to expose it some other way, perhaps some `func as(_ type: T) -> T?`? The reason is that when we get "some distributed actor" we need to check if we understand the identity quite often (in the cluster I hit this a lot).

Second thought: If we adopted the well-typed `Identity` i guess this would become unnecessary since we would only work with "our" actors hm... I do keep wondering if it isn't too limiting to erase without being able to pull out the underlying identity though hm...

Summary: Not sure yet 🤔